From 495ed57f10c184daf5b68f347dac0be01ca6841b Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Sat, 5 Mar 2016 15:31:04 +0100
Subject: [PATCH 101/108] UTIL: Fix warning misleading-indentation
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Warnings are emited from macro generated code in dlinklist.h
e.g.
src/ldb_modules/memberof.c:4209:13: error: statement is indented as if it were
    guarded by... [-Werror=misleading-indentation]
             DLIST_DEMOTE(ctx->group_list, grp, struct mbof_member *);
             ^~~~~~~~~~~~
src/ldb_modules/memberof.c:4209:13: note: ...this ‘if’ clause, but it is not
src/ldb_modules/memberof.c: In function ‘mbof_member_update’:
src/ldb_modules/memberof.c:4305:9: error: statement is indented as if it were
    guarded by... [-Werror=misleading-indentation]
         DLIST_PROMOTE(ctx->group_list, mem);
         ^~~~~~~~~~~~~
src/ldb_modules/memberof.c:4305:9: note: ...this ‘if’ clause, but it is not
src/ldb_modules/memberof.c: In function ‘mbof_rcmp_update’:
src/ldb_modules/memberof.c:4408:9: error: statement is indented as if it were
    guarded by... [-Werror=misleading-indentation]
         DLIST_REMOVE(ctx->user_list, x);
         ^~~~~~~~~~~~
src/util/crypto/nss/nss_obfuscate.c: In function ‘sss_password_decrypt’:
src/util/crypto/nss/nss_obfuscate.c:419:5: error: statement is indented
    as if it were guarded by... [-Werror=misleading-indentation]
     SAFEALIGN_COPY_UINT16_CHECK(&meth, obfbuf+p, obflen, &p);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/python/pyhbac.c: In function ‘PyInit_pyhbac’:
src/python/pyhbac.c:1987:5: error: statement is indented as if it were
    guarded by... [-Werror=misleading-indentation]
     TYPE_READY(m, pyhbac_hbacrule_type, "HbacRule");
     ^~~~~~~~~~
src/python/pyhbac.c:1987:5: note: ...this ‘if’ clause, but it is not

Reviewed-by: Pavel Březina <pbrezina@redhat.com>
Reviewed-by: Michal Židek <mzidek@redhat.com>
(cherry picked from commit c6278b2fa4a7ea389ed4086b2def16e0e6cbb184)
(cherry picked from commit 19580e01096f8a57e37414ef3f5f2d0a6528da34)
---
 src/util/dlinklist.h      | 24 ++++++++++++++++++------
 src/util/sss_python.h     |  5 +++--
 src/util/util_safealign.h |  6 +++---
 3 files changed, 24 insertions(+), 11 deletions(-)

diff --git a/src/util/dlinklist.h b/src/util/dlinklist.h
index e8490496c4d8a8018edbeec776b2f09d2066236c..4f6aef830e914c22654970081263d43461c1750f 100644
--- a/src/util/dlinklist.h
+++ b/src/util/dlinklist.h
@@ -43,12 +43,20 @@ do { \
 do { \
     if ((p) == (list)) { \
         (list) = (p)->next; \
-        if (list) (list)->prev = NULL; \
+        if (list) { \
+            (list)->prev = NULL; \
+        } \
     } else { \
-        if ((p)->prev) (p)->prev->next = (p)->next; \
-        if ((p)->next) (p)->next->prev = (p)->prev; \
+        if ((p)->prev) { \
+            (p)->prev->next = (p)->next; \
+        } \
+        if ((p)->next) { \
+            (p)->next->prev = (p)->prev; \
+        } \
+    } \
+    if ((p) != (list)) { \
+        (p)->next = (p)->prev = NULL; \
     } \
-    if ((p) != (list)) (p)->next = (p)->prev = NULL; \
 } while (0)
 
 /* promote an element to the top of the list */
@@ -85,7 +93,9 @@ do { \
         p->prev = el; \
         p->next = el->next; \
         el->next = p; \
-        if (p->next) p->next->prev = p; \
+        if (p->next) { \
+            p->next->prev = p; \
+        } \
     } \
 } while (0)
 
@@ -128,7 +138,9 @@ do { \
         (list2)->prev = (el); \
         tmp->next = (el)->next; \
         (el)->next = (list2); \
-        if (tmp->next != NULL) tmp->next->prev = tmp; \
+        if (tmp->next != NULL) { \
+            tmp->next->prev = tmp; \
+        } \
     } \
 } while (0);
 
diff --git a/src/util/sss_python.h b/src/util/sss_python.h
index 7e2bac33656dcbac91bb4f4d32ec9fbc44bb4e52..b3fdaad646af430de6ad8fde7dd66efa608aa89a 100644
--- a/src/util/sss_python.h
+++ b/src/util/sss_python.h
@@ -31,8 +31,9 @@ sss_exception_with_doc(char *name, char *doc, PyObject *base, PyObject *dict);
 
 /* Convenience macros */
 #define TYPE_READY(module, type, name) do {         \
-    if (PyType_Ready(&type) < 0)                    \
-	MODINITERROR;                               \
+    if (PyType_Ready(&type) < 0) {                  \
+        MODINITERROR;                               \
+    }                                               \
     Py_INCREF(&type);                               \
     PyModule_AddObject(module,                      \
                        discard_const_p(char, name), \
diff --git a/src/util/util_safealign.h b/src/util/util_safealign.h
index ba216f6063a34524c40f961115d79b40ec4e1641..b1c9f8a0c11f4d13fca885aa4e28e7c4750f37fe 100644
--- a/src/util/util_safealign.h
+++ b/src/util/util_safealign.h
@@ -103,19 +103,19 @@ safealign_memcpy(void *dest, const void *src, size_t n, size_t *counter)
  * would excceed len. */
 #define SAFEALIGN_COPY_UINT32_CHECK(dest, src, len, pctr) do { \
     if ((*(pctr) + sizeof(uint32_t)) > (len) || \
-        SIZE_T_OVERFLOW(*(pctr), sizeof(uint32_t))) return EINVAL; \
+        SIZE_T_OVERFLOW(*(pctr), sizeof(uint32_t))) { return EINVAL; } \
     safealign_memcpy(dest, src, sizeof(uint32_t), pctr); \
 } while(0)
 
 #define SAFEALIGN_COPY_INT32_CHECK(dest, src, len, pctr) do { \
     if ((*(pctr) + sizeof(int32_t)) > (len) || \
-        SIZE_T_OVERFLOW(*(pctr), sizeof(int32_t))) return EINVAL; \
+        SIZE_T_OVERFLOW(*(pctr), sizeof(int32_t))) { return EINVAL; } \
     safealign_memcpy(dest, src, sizeof(int32_t), pctr); \
 } while(0)
 
 #define SAFEALIGN_COPY_UINT16_CHECK(dest, src, len, pctr) do { \
     if ((*(pctr) + sizeof(uint16_t)) > (len) || \
-        SIZE_T_OVERFLOW(*(pctr), sizeof(uint16_t))) return EINVAL; \
+        SIZE_T_OVERFLOW(*(pctr), sizeof(uint16_t))) { return EINVAL; } \
     safealign_memcpy(dest, src, sizeof(uint16_t), pctr); \
 } while(0)
 
-- 
2.7.3

